workspace/src/main/kotlin/com/cosmotech/workspace/service/WorkspaceServiceImpl.kt

```diff
@@ -154,9 +155,9 @@ internal class WorkspaceServiceImpl(
               " please refer to the appropriate security endpoints to perform this maneuver")
     }
     return if (hasChanged) {
-      workspaceRepository.save(existingWorkspace)
+      checkReadSecurity(workspaceRepository.save(existingWorkspace))
```

Comment:
Same remark here, if you're able to update something, you have all permission on it